src/ooni/utils.cpp

```diff
@@ -0,0 +1,57 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "src/ooni/utils.hpp"
+
+namespace mk {
+namespace ooni {
+
+ErrorOr<Json> geoip(std::string ip, std::string path_country,
+                    std::string path_asn) {
+    Json json;
+    GeoIP *gi;
+    GeoIPLookup gl;
+
+    gi = GeoIP_open(path_country.c_str(), GEOIP_MEMORY_CACHE);
+    if (gi == nullptr) {
+        GeoIP_delete(gi);
```

Comment:
Here you should not pass `nullptr` to `GeoIP_delete()`. No need to free resources in this error path because we are in the case where no resources have actually been allocated.